conans/client/graph/compatibility.py

```diff
@@ -29,7 +29,9 @@ def compatibility(conanfile):
 
 def cppstd_compat(conanfile):
     # It will try to find packages with all the cppstd versions
-
+    if not conanfile.conf_info.get("user.conan.plugins:compatibility_cppstd", check_type=bool,
```

Comment:
Note it reads directly ``conf_info``, not ``conf``

The specifics of ``user.conan.plugins:compatibility_cppstd`` are to be discussed:
- Do we want a builtin ``tools.conan ....`` (tools are more for recipe tools doesn't seem the best match)
- Do we want a ``core.plugins:compatibility_cppstd``? Maybe sounds better

The ``conf_info`` is only propagated to consumers ``conf`` in the case of direct ``tool_requires``:
- Is it an issue that consumers of this recipe gets the ``compatibility_cppstd`` information from the dependency? Doesn't seem harmful, consumers will mostly ignore it, not an issue if you don't read it

Reply:
Changed to ``extension_properties`` instead of conf